server/src/main/java/org/elasticsearch/index/IndexMode.java

```diff
@@ -194,6 +202,25 @@ public DocumentDimensions buildDocumentDimensions() {
         public boolean shouldValidateTimestamp() {
             return true;
         }
+
+        /**
+         * For a time series index we just support 'fixed_interval' aggregations.
```

Comment:
I think I might have missed something. I thought we only wanted to limit to fixed interval for the rolled up indices. Time series indices are supposed to support all the things. I think. I certainly could be forgetting things, but that was my memory.

Reply:
good catch, this was also my understanding

Reply:
What about the UTC/non-UTC check?

Reply:
Same for UTC/non-UTC by my understanding. Not sure whether it's right or not, but in Kibana we currently check this via the `meta.fixed_interval` property on the field: https://github.com/elastic/kibana/blob/9799dbba27c5baf594357eae0bbfc79b4e7da77c/src/plugins/data_views/server/fetcher/lib/field_capabilities/field_caps_response.ts#L145

Reply:
> I don't get what is wrong here...do you mean this should fail only if the date histogram aggregation runs on a rolled-up index? And not on every time series index? If that is what I missed then I have no idea how do we know if an index is a result of a rollup operation or not. Maybe we have some meta-data? I will check.

It should fail only if the `date_histogram` agg is executed on a rolled up index. "Raw" time-series indices support any `date_histogram` interval. 
Rollup indices contain the `IndexMetadata.INDEX_ROLLUP_SOURCE_NAME` setting in their index settings. And the `@timestamp` field should contain the `fixed_interval` property in the field mapping `meta`. See https://github.com/elastic/elasticsearch/blob/03f3c8119ac1e7bb7034be1f7c4f3355fec4b300/x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java#L440-L453